packages/orchestration/src/orchestration-client.ts

```diff
@@ -42,7 +45,9 @@ export class OrchestrationClient {
    */
   constructor(
     private config: OrchestrationModuleConfig | string,
-    private deploymentConfig?: ResourceGroupConfig,
+    private deploymentConfig?: Partial<
```

Comment:
[q] Why `Partial` ? deploymentConfig is already optional. `ResourceGroupConfig | DeploymentIdConfig` or `Xor<ResourceGroupConfig, DeploymentIdConfig>` might make more sense.

Reply:
`ResourceGroupConfig | DeploymentIdConfig` should work as eventually both of them are optional. Xor maybe not as they are not mutual exclusive?

Reply:
Yeah true, the header might still be needed.

Reply:
This still needs to be fixed. Change it to `private deploymentConfig?: ResourceGroupConfig | DeploymentIdConfig`